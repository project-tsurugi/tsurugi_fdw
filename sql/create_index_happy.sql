/* Test setup: DDL of the Tsurugi */
SELECT tg_execute_ddl('
    CREATE TABLE t3_create_index (
        c1 INTEGER NOT NULL PRIMARY KEY
    )
', 'tsurugidb');
SELECT tg_execute_ddl('
    CREATE INDEX t3_c1_secondary_index ON t3_create_index (c1)
', 'tsurugidb');
SELECT tg_execute_ddl('
    CREATE TABLE t4_create_index (
        c1 INTEGER NOT NULL PRIMARY KEY,
        c2 BIGINT,
        c3 DOUBLE PRECISION
    )
', 'tsurugidb');
SELECT tg_execute_ddl('
    CREATE INDEX t4_c3_secondary_index ON t4_create_index (c3)
', 'tsurugidb');

/* Test setup: DDL of the PostgreSQL */
CREATE FOREIGN TABLE t3_create_index (
    c1 INTEGER NOT NULL
) SERVER tsurugidb;
CREATE FOREIGN TABLE t4_create_index (
    c1 INTEGER NOT NULL,
    c2 BIGINT,
    c3 DOUBLE PRECISION
) SERVER tsurugidb;

/* DML */
SELECT * FROM t3_create_index ORDER BY c1;
INSERT INTO t3_create_index (c1) VALUES (1);
SELECT * FROM t3_create_index ORDER BY c1;

SELECT * FROM t4_create_index ORDER BY c1;
INSERT INTO t4_create_index (c1, c2, c3) VALUES (1, 100, 1.1);
INSERT INTO t4_create_index (c1, c2, c3) VALUES (2, 200, 2.2);
SELECT * FROM t4_create_index ORDER BY c1;

SELECT * FROM t3_create_index ORDER BY c1;
INSERT INTO t3_create_index (c1) VALUES (10);
INSERT INTO t3_create_index (c1) VALUES (100);
SELECT * FROM t3_create_index ORDER BY c1;

SELECT * FROM t4_create_index ORDER BY c1;
UPDATE t4_create_index SET c2 = c2+10;
UPDATE t4_create_index SET c3 = c3+1.1 WHERE c2 = 110;
SELECT * FROM t4_create_index ORDER BY c1;

SELECT * FROM t3_create_index ORDER BY c1;
DELETE FROM t3_create_index WHERE c1 = 10;
SELECT * FROM t3_create_index ORDER BY c1;

/* Test teardown: DDL of the PostgreSQL */
DROP FOREIGN TABLE t3_create_index;
DROP FOREIGN TABLE t4_create_index;

/* Test teardown: DDL of the Tsurugi */
SELECT tg_execute_ddl('DROP INDEX t3_c1_secondary_index', 'tsurugidb');
SELECT tg_execute_ddl('DROP TABLE t3_create_index', 'tsurugidb');
SELECT tg_execute_ddl('DROP INDEX t4_c3_secondary_index', 'tsurugidb');
SELECT tg_execute_ddl('DROP TABLE t4_create_index', 'tsurugidb');
