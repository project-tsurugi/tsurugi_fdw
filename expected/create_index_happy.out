/* Test setup: DDL of the Tsurugi */
SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE t3_create_index (
        c1 INTEGER NOT NULL PRIMARY KEY
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE INDEX t3_c1_secondary_index ON t3_create_index (c1)
');
 tg_execute_ddl 
----------------
 CREATE INDEX
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE TABLE t4_create_index (
        c1 INTEGER NOT NULL PRIMARY KEY,
        c2 BIGINT,
        c3 DOUBLE PRECISION
    )
');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', '
    CREATE INDEX t4_c3_secondary_index ON t4_create_index (c3)
');
 tg_execute_ddl 
----------------
 CREATE INDEX
(1 row)

/* Test setup: DDL of the PostgreSQL */
CREATE FOREIGN TABLE t3_create_index (
    c1 INTEGER NOT NULL
) SERVER tsurugidb;
CREATE FOREIGN TABLE t4_create_index (
    c1 INTEGER NOT NULL,
    c2 BIGINT,
    c3 DOUBLE PRECISION
) SERVER tsurugidb;
/* DML */
SELECT * FROM t3_create_index ORDER BY c1;
 c1 
----
(0 rows)

INSERT INTO t3_create_index (c1) VALUES (1);
SELECT * FROM t3_create_index ORDER BY c1;
 c1 
----
  1
(1 row)

SELECT * FROM t4_create_index ORDER BY c1;
 c1 | c2 | c3 
----+----+----
(0 rows)

INSERT INTO t4_create_index (c1, c2, c3) VALUES (1, 100, 1.1);
INSERT INTO t4_create_index (c1, c2, c3) VALUES (2, 200, 2.2);
SELECT * FROM t4_create_index ORDER BY c1;
 c1 | c2  | c3  
----+-----+-----
  1 | 100 | 1.1
  2 | 200 | 2.2
(2 rows)

SELECT * FROM t3_create_index ORDER BY c1;
 c1 
----
  1
(1 row)

INSERT INTO t3_create_index (c1) VALUES (10);
INSERT INTO t3_create_index (c1) VALUES (100);
SELECT * FROM t3_create_index ORDER BY c1;
 c1  
-----
   1
  10
 100
(3 rows)

SELECT * FROM t4_create_index ORDER BY c1;
 c1 | c2  | c3  
----+-----+-----
  1 | 100 | 1.1
  2 | 200 | 2.2
(2 rows)

UPDATE t4_create_index SET c2 = c2+10;
UPDATE t4_create_index SET c3 = c3+1.1 WHERE c2 = 110;
SELECT * FROM t4_create_index ORDER BY c1;
 c1 | c2  | c3  
----+-----+-----
  1 | 110 | 2.2
  2 | 210 | 2.2
(2 rows)

SELECT * FROM t3_create_index ORDER BY c1;
 c1  
-----
   1
  10
 100
(3 rows)

DELETE FROM t3_create_index WHERE c1 = 10;
SELECT * FROM t3_create_index ORDER BY c1;
 c1  
-----
   1
 100
(2 rows)

/* Test teardown: DDL of the PostgreSQL */
DROP FOREIGN TABLE t3_create_index;
DROP FOREIGN TABLE t4_create_index;
/* Test teardown: DDL of the Tsurugi */
SELECT tg_execute_ddl('tsurugidb', 'DROP INDEX t3_c1_secondary_index');
 tg_execute_ddl 
----------------
 DROP INDEX
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE t3_create_index');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP INDEX t4_c3_secondary_index');
 tg_execute_ddl 
----------------
 DROP INDEX
(1 row)

SELECT tg_execute_ddl('tsurugidb', 'DROP TABLE t4_create_index');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

